espnet2/asr/transducer/beam_search_transducer.py

```diff
@@ -244,18 +261,58 @@ def default_beam_search(self, enc_out: torch.Tensor) -> List[Hypothesis]:
             nbest_hyps: N-best hypothesis.
 
         """
+        logging.info("decoder input length: " + str(enc_out.shape[0]))
         beam = min(self.beam_size, self.vocab_size)
         beam_k = min(beam, (self.vocab_size - 1))
 
         dec_state = self.decoder.init_state(1)
 
         kept_hyps = [Hypothesis(score=0.0, yseq=[self.blank_id], dec_state=dec_state)]
         cache = {}
+        cache_lm = {}
```

Comment:
Simple curiosity, why did you define the cache as dict with tuple keys? Also, we could maybe sync both caches.

Reply:
Since `yseq` is a list, which is unhashable by a dictionary, I converted the list into a tuple. If this is not a good way, please let me know a better alternate way.

Reply:
I'm not sure what the best practice is here. For cache, we use a named dict (Dict[str, Any]).

Honestly, I don't think it will change a thing in case of default beam search. Efficiency becomes important when we talk about the batched search process, when splitting and rearranging processed and unprocess inputs for decoder or LM computation.